extensions/wikia/BannerNotifications/js/BannerNotifications.js

```diff
@@ -0,0 +1,269 @@
+/*
+ * Handles the color-coded notification messages that generally appear
+ * at the top of the screen or inside a modal.
+ * Use like:
+ * BannerNotifications.show('Some success message', 'confirm')
+ * BannerNotifications.show('Some error message', 'error', $('.myDiv'), 3000)
+ */
+define('BannerNotifications', ['jquery', 'wikia.window'], function ($, window) {
+	'use strict';
+
+	var defaultTimeout = 10000,
+		types = {
+			'notify': 'blue',
+			'confirm': 'green',
+			'error': 'red',
+			'warn': 'yellow'
+		},
+		classes = Object.keys(types).join(' '),
+		pageContainer,
+		wikiaHeader,
+		headerHeight,
+		modal,
+		isModal;
+
+	/**
+	 * Constructs jQuery element with the notification
+	 * @param {String} content
+	 * @param {String} [type]
+	 * @returns {jQuery}
+	 */
+	function createMarkup(content, type) {
```

Comment:
It would be nice to create a mustache template and import it via `grunt-mustache`. https://one.wikia-inc.com/wiki/Engineering/Grunt-Mustache